sdk/core/Azure.Core/src/Internal/BitVector640.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Internal;
+
+/// <summary>
+/// This type effectively stores 640 bool values, but compresses their storage
+/// into ten ulongs, where each bit of the ulong represents a single bool value.
+///
+/// It exposes a public indexer so that the bool values can be accessed as a
+/// standard .NET collection API.
+///
+/// It is used in System.ClientModel and Azure.Core to implement response
+/// classifiers, where each true bit represents a status code that the
+/// classifier considers a success code.
+/// </summary>
+internal struct BitVector640
```

Comment:
Copied in from System.ClientModel/Internal.
Removing duplication across Azure.Core and System.ClientModel is tracked by https://github.com/Azure/azure-sdk-for-net/issues/41035